services/normalDocSvc.js

```diff
@@ -5,197 +5,198 @@
+      var explainable = function(doc, explainJson) {
+
+        var simplerExplain = null;// explainSvc.createExplain(explainJson);
+        var hotMatches = null;//simplerExplain.vectorize();
+        var matchDetails = null;
+
+        var initExplain = function() {
+          if (!simplerExplain) {
+            simplerExplain = explainSvc.createExplain(explainJson);
+            hotMatches = simplerExplain.vectorize();
+            matchDetails = simplerExplain.matchDetails();
+          }
+        };
+
+        doc.explain = function() {
+          initExplain();
+          return simplerExplain;
+        };
+
+        doc.hotMatches = function() {
+          initExplain();
+          return hotMatches;
+        };
+
+        doc.matchDetails = function() {
+          initExplain();
+          return matchDetails;
+        };
+
+        var hotOutOf = [];
+        var lastMaxScore = -1;
+        doc.hotMatchesOutOf = function(maxScore) {
+          initExplain();
+          if (maxScore !== lastMaxScore) {
+            hotOutOf.length = 0;
+          }
+          lastMaxScore = maxScore;
+          if (hotOutOf.length === 0) {
+            angular.forEach(hotMatches.vecObj, function(value, key) {
+              var percentage = ((0.0 + value) / maxScore) * 100.0;
+              hotOutOf.push({description: key, metadata: matchDetails[key], percentage: percentage});
+            });
+            hotOutOf.sort(function(a,b) {return b.percentage - a.percentage;});
+          }
+          return hotOutOf;
+        };
+
+        doc.score = function() {
+          initExplain();
+          return simplerExplain.contribution();
+        };
+        return doc;
       };
 
-      var hotOutOf = [];
-      var lastMaxScore = -1;
-      doc.hotMatchesOutOf = function(maxScore) {
-        initExplain();
-        if (maxScore !== lastMaxScore) {
-          hotOutOf.length = 0;
+      var getDocExplain = function(doc, nDoc) {
+        var explJson = doc.explain(nDoc.id);
+        if (explJson === null) {
+          if (doc.source().hasOwnProperty('id')) {
+            return doc.explain(doc.source().id);
+          }
         }
-        lastMaxScore = maxScore;
-        if (hotOutOf.length === 0) {
-          angular.forEach(hotMatches.vecObj, function(value, key) {
-            var percentage = ((0.0 + value) / maxScore) * 100.0;
-            hotOutOf.push({description: key, metadata: matchDetails[key], percentage: percentage});
-          });
-          hotOutOf.sort(function(a,b) {return b.percentage - a.percentage;});
+        return explJson;
+      };
+
+      this.createNormalDoc = function(fieldSpec, doc, altExplainJson) {
+        var nDoc = new NormalDoc(fieldSpec, doc);
+        var explJson;
+        if (altExplainJson) {
+          explJson = altExplainJson;
+        } else {
+          explJson = getDocExplain(doc, nDoc);
         }
-        return hotOutOf;
+        return this.snippetDoc(this.explainDoc(nDoc, explJson));
```

Comment:
Resolved as rendering bug in Github. There was no logic change, just formatting. Confirmed with @ychaker